pandas/core/indexes/accessors.py

```diff
@@ -20,81 +21,44 @@
 from pandas.core.algorithms import take_1d
 
 
-def is_datetimelike(data):
-    """
-    return a boolean if we can be successfully converted to a datetimelike
-    """
-    try:
-        maybe_to_datetimelike(data)
-        return True
-    except (Exception):
-        pass
-    return False
+class Properties(PandasDelegate, PandasObject, NoNewAttributesMixin):
 
+    def __init__(self, data, orig):
+        if not isinstance(data, ABCSeries):
+            raise TypeError("cannot convert an object of type {0} to a "
+                            "datetimelike index".format(type(data)))
 
-def maybe_to_datetimelike(data, copy=False):
-    """
-    return a DelegatedClass of a Series that is datetimelike
-      (e.g. datetime64[ns],timedelta64[ns] dtype or a Series of Periods)
-    raise TypeError if this is not possible.
+        self.values = data
+        self.orig = orig
+        self.name = getattr(data, 'name', None)
+        self.index = getattr(data, 'index', None)
+        self._freeze()
 
-    Parameters
-    ----------
-    data : Series
-    copy : boolean, default False
-           copy the input data
+    def _get_values(self):
```

Comment:
generally would like these to be non-private

Reply:
I think it's best for methods and attributes on the accessor to be private. Imagine if we wanted to delegate a method like `.get_values`, so `foo.bar.get_values`. This would break the accessor, since its `get_values` would be broken.